File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/security/ProjectPermission.java
package net.nemerosa.ontrack.model.security;

import lombok.Data;

/**
 * Definition of a permission on a project: a {@link net.nemerosa.ontrack.model.security.PermissionTarget}
 * gets associated with a {@link net.nemerosa.ontrack.model.security.ProjectRole}.
 */
@Data
public class ProjectPermission {

    private final PermissionTarget target;
    private final ProjectRole role;

}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/security/ProjectPermission.kt
package net.nemerosa.ontrack.model.security

import net.nemerosa.ontrack.model.structure.ID

/**
 * Definition of a permission on a project: a [net.nemerosa.ontrack.model.security.PermissionTarget]
 * gets associated with a [net.nemerosa.ontrack.model.security.ProjectRole].
 */
class ProjectPermission(val projectId: ID, val target: PermissionTarget, val role: ProjectRole)